src/Compilers/CSharp/Test/Emit3/ExtensionTypeTests.cs

```diff
@@ -42696,4 +42742,1408 @@ .maxstack 8
     // PROTOTYPE(roles): Ensure instance events are never treated as WINRT events.
 
     // PROTOTYPE(roles): Test consumption of instance APIs from VB as static APIs. Also check behavior of legacy C# compilers.
+
+    [Fact]
+    public void InstanceMethod_Metadata_ConsumptionFromPreviousVersion_01()
+    {
+        // public implicit extension E for C
+        // {
+        //     public void Method()
+        //     {
+        //         this.Increment();
+        //     }
+        // }
+        // 
+        // public class C
+        // {
+        //     public int F;
+        // 
+        //     public void Increment()
+        //     {
+        //         F++;
+        //     }
+        // }
+        var ilSource = """
+.class public sequential ansi sealed beforefieldinit E
+	extends [mscorlib]System.ValueType
+{
+    .field private class C '<UnderlyingInstance>$'
+    .custom instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = (
+    	01 00 00 00
+    )
+
+	.method public hidebysig static 
+		void Method (
+			class C modopt([mscorlib]System.Runtime.CompilerServices.ExtensionAttribute) '<>4__this'
+		) cil managed 
+	{
+		.maxstack 8
+		IL_0000: ldarg.0
+		IL_0001: callvirt instance void C::Increment()
+		IL_0006: ret
+	}
+
+	.method public hidebysig static 
+		void '<ImplicitExtension>$' (
+			class C ''
+		) cil managed 
+	{
+		.maxstack 8
+		IL_0000: ret
+	}
+}
+
+.class public auto ansi beforefieldinit C
+    extends [mscorlib]System.Object
+{
+    .field public int32 F
+
+    .method public hidebysig 
+    	instance void Increment () cil managed 
+    {
+    	.maxstack 8
+    	IL_0000: ldarg.0
+    	IL_0001: ldarg.0
+    	IL_0002: ldfld int32 C::F
+    	IL_0007: ldc.i4.1
+    	IL_0008: add
+    	IL_0009: stfld int32 C::F
+    	IL_000e: ret
+    }
+
+    .method public hidebysig specialname rtspecialname 
+    	instance void .ctor () cil managed 
+    {
+    	.maxstack 8
+    	IL_0000: ldarg.0
+    	IL_0001: call instance void [mscorlib]System.Object::.ctor()
+    	IL_0006: ret
+    }
+}
+""";
+
+#if !PREVIOUSVERSIONOFCOMPILER
```

Comment:
What's `PREVIOUSVERSIONOFCOMPILER`? It's not defined anywhere as far as I can see.

Reply:
> It's not defined anywhere as far as I can see.

That is the intent. The branch that is not taken reflects behavior of the compiler in `main`. Obviously, it is going to fail in `roles`. The tests can be run manually in `main' by copy/pasting them into a file and defining the conditional compilation symbol.

Reply:
Consider leaving a comment on one of those directives to provide context.

Reply:
> Consider leaving a comment on one of those directives to provide context.

I will do this in a separate PR